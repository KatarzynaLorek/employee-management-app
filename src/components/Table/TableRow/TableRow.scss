.row {
  display: flex;
  flex-direction: row;
  border-top: 1px solid #ddd;

  &--title {
    border-bottom: 1px solid #ccc;
    border-top: 1px solid #ddd;
    text-transform: uppercase;
    font-weight: 700;
  }

  &__part {
    flex-basis: 20%;
    height: 70px;
    text-align: center;
    line-height: 70px;
  }

  &__label {
    display: none;
  }

  &__button {
    margin: 0 5px;
    width: 40%;
    height: 40px;
    border-radius: 9px;
    background-color: #fff;
    text-transform: uppercase;
    line-height: 40px;
    transition: 0.1s linear;
    cursor: pointer;

    &--update {
      border: 2px solid orange;

      &:hover {
        background-color: orange;
        color: white;
      }
    }

    &--delete {
      border: 2px solid red;

      &:hover {
        background-color: red;
        color: white;
      }
    }
  }
}

@media (max-width: 767px) {
  .row {
    flex-direction: column;
    padding: 10px 0 10px 30px;
    &--title {
      display: none;
    }

    &__part {
      display: flex;
      flex-direction: row;
      line-height: 30px;
      text-align: left;
      padding-bottom: 7px;
    }

    &__label {
      display: block;
      width: 100px;
      font-weight: 700;
      margin-right: 20px;
    }

    &__button {
      max-width: 100px;
      height: 30px;
      line-height: 30px;
    }
  }
} ;
