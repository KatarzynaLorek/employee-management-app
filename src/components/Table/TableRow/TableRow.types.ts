import { IResponseObject } from '../../../types/responses';

export enum TableRowVariant {
  title,
  normal,
}

export interface ITableRow {
  variant: TableRowVariant;
  employeeData?: IResponseObject;
  handleUpdateButtonClick?: (employee: IResponseObject) => void;
  handleDeleteButtonClick?: (employee: IResponseObject) => void;
}
