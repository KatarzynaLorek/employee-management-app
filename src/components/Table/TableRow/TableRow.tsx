import React from 'react';
import { useAppDispatch } from '../../../hooks/reduxHooks';
import { removeEmployee } from '../../../store/actions/actions';
import { IResponseObject } from '../../../types/responses';
import './TableRow.scss';

interface TableRowProps {
  employeeData?: IResponseObject;
  handleClick: ((employee?: IResponseObject) => void) | (() => void);
}

const TableRow = ({ employeeData, handleClick }: TableRowProps): JSX.Element => {
  const dispatch = useAppDispatch();
  return (
    <div className={employeeData ? 'row' : 'row row--title'}>
      <div className="row__part">
        <p className="row__label">First Name</p>
        <p>{employeeData ? employeeData.firstName : 'First Name'}</p>
      </div>
      <div className="row__part">
        <p className="row__label">Last Name</p>
        <p>{employeeData ? employeeData.lastName : 'Last Name'}</p>
      </div>
      <div className="row__part">
        <p className="row__label">Department</p>
        <p>{employeeData ? employeeData.department : 'Department'}</p>
      </div>
      <div className="row__part">
        <p className="row__label">Position</p>
        <p>{employeeData ? employeeData.position : 'Position'}</p>
      </div>
      <div className="row__part">
        {employeeData ? (
          <>
            <button
              onClick={() => (employeeData ? handleClick(employeeData) : null)}
              className="row__button row__button--update"
            >
              Update
            </button>
            <button
              className="row__button row__button--delete"
              onClick={() => {
                dispatch(removeEmployee(employeeData.id));
              }}
            >
              delete
            </button>
          </>
        ) : (
          'Actions'
        )}
      </div>
    </div>
  );
};

export default TableRow;
