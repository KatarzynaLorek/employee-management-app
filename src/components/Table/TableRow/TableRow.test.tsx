import React from 'react';
import { getByText, render } from '@testing-library/react';
import TableRow from './TableRow';

describe('TableRow component', () => {
  it('renders properly', () => {
    const { getByText } = render(<TableRow />);
    expect(getByText('Name')).toBeInTheDocument();
  });

  it('renders title row', () => {
    const { getByText } = render(<TableRow isTitle />);
    expect(getByText('Actions')).toBeInTheDocument;
  });

  it('renders normal row', () => {
    const { getByText, container } = render(<TableRow />);
    expect(getByText('Update')).toBeInTheDocument;
    expect(container).not.toHaveTextContent('Actions');
  });
});
