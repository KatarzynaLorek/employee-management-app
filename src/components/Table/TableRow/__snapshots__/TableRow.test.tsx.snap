// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow component renders normal row 1`] = `
<body>
  <div>
    <div
      class="row"
    >
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          First Name
        </p>
        <p>
          mockFirstName
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Last Name
        </p>
        <p>
          mockLastname
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Department
        </p>
        <p>
          IT
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Position
        </p>
        <p>
          junior
        </p>
      </div>
      <div
        class="row__part"
      >
        <button
          class="row__button--update"
        >
          Update
        </button>
        <button
          class="row__button--delete"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`TableRow component renders title row 1`] = `
<body>
  <div>
    <div
      class="row--title"
    >
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          First Name
        </p>
        <p>
          First Name
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Last Name
        </p>
        <p>
          Last Name
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Department
        </p>
        <p>
          Department
        </p>
      </div>
      <div
        class="row__part"
      >
        <p
          class="row__label"
        >
          Position
        </p>
        <p>
          Position
        </p>
      </div>
      <div
        class="row__part"
      >
        Actions
      </div>
    </div>
  </div>
</body>
`;
